tsdb/wal/watcher.go

```diff
@@ -197,14 +213,28 @@ func (w *Watcher) Stop() {
 
 func (w *Watcher) loop() {
 	defer close(w.done)
+	startSegment := -1
+
+	// look for a checkpoint file
+	fname := filepath.Join(w.tsdbDir, "remote", w.name, "checkpoint")
```

Comment:
Putting `remote` into a wal specific file doesn't seem ideal. Could it just be `watcher` instead? Or specified when creating a Watcher?